chai-immutable.js

```diff
@@ -437,10 +438,15 @@ module.exports = function (chai, utils) {
    */
 
   assert.equal = function (actual, expected) {
+    /*
+     * It seems like we shouldn't actually need this check, however,
+     * `assert.equal` actually behaves differently than its BDD counterpart!
+     * Namely, the BDD version is strict while the "assert" one isn't.
+     */
     if (actual instanceof Collection) {
       return new Assertion(actual).equal(expected);
     }
-    else return assert.equal;
+    else return originalEqual(actual, expected);
```

Comment:
You nailed it, actually! The issue isn't so much that I didn't call the original assertion, but that I was returning a function, which never throws anything. And of course, not storing the original assertion somewhere triggers an infinite loop.
Good job!

That's one more proof we need more tests...